lib/internal/repl.js

```diff
@@ -17,16 +20,18 @@ function replStart() {
   return REPL.start.apply(REPL, arguments);
 }
 
-function createRepl(env, cb) {
+function createRepl(env, attemptPersistentHistory, cb) {
   const opts = {
     ignoreUndefined: false,
-    terminal: process.stdout.isTTY,
     useGlobal: true
   };
 
   if (parseInt(env.NODE_NO_READLINE)) {
     opts.terminal = false;
   }
+  if (parseInt(env.NODE_FORCE_READLINE)) {
```

Comment:
It shouldn't be a problem in this case – we're just detecting whether the number is "0" or not.